cmd/minishift/cmd/config/get.go

```diff
@@ -45,10 +47,15 @@ var configGetCmd = &cobra.Command{
 
 func init() {
 	ConfigCmd.AddCommand(configGetCmd)
+	configGetCmd.Flags().BoolVar(&global, "global", false, "Get the value of a configuration property in the global configuration file.")
 }
 
 func get(name string) (string, error) {
-	m, err := ReadConfig()
+	confFile := constants.ConfigFile
+	if global {
+		confFile = constants.GlobalConfigFile
+	}
+	m, err := config.ReadViperConfig(confFile)
```

Comment:
Shouldn't this be called `ReadInstanceConfig` or so?

Reply:
I renamed all the viper related things to mention that is related to Viper. `ReadConfg` means read any config but actually, it can read the config which viper create using `m config set`.

Reply:
but do we care about the technical implementation in a name?
I believe the naming is confusing and does not help a newcomer.

one of the configs is libmachine config, and the other is minishift instance config?

Reply:
@gbraad I put the name only to make difference for different config files like we have for instance then allinstance so it will make bit easy to understand which config we are reading/writing or referring.

Reply:
I do not find `ReadViperConfig` is doing anything specific for Viper. So IMO just `ReadConfig` is better.

Reply:
@LalatenduMohanty this is dealing with the config which Viper is using.

Reply:
https://github.com/minishift/minishift/blob/master/cmd/minishift/cmd/root.go#L289-L291 here you can see how viper consume it.